File: frontend/lib/capacidades.ts
// Catálogo de capacidades (ERP+CRM) — espejo de shared/rubros/capacidades.py (backend).
// Mantener ambos en sync. 22 capacidades = 20 del artifact "Sidebar · 3 propuestas para
// 100+ rubros" + 2 nuevas ERP (compras, produccion) decididas en Fase 0.
// El campo `capacidades` STORED por rubro (lib/rubros.ts) es el canónico (curado,
// Fase 2). La derivación §3 se retiró junto con `modulos` en la Etapa B; los modos
// de dato (recurso/variantes/precioMedida) viven como flags del RubroDef.

// Orden canónico (CAP_ORDER): las listas `capacidades` de cada rubro se guardan en
// este orden en AMBOS SSOT (Py y TS) para que el espejo sea determinista.
export const CAPACIDADES = [
  "catalogo",
  "agenda",
  "cotizador",
  "pedidos",
  "pagos",
  "delivery",
  "suscripciones",
  "soporte",
  "proyectos",
  "inventario",
  "compras",
  "produccion",
  "expedientes",
  "contratos",
  "cursos",
  "arriendos",
  "facturacion",
  "fidelizacion",
  "campanas",
  "terreno",
  "mesas",
  "consentimientos",
] as const;

export type Capacidad = (typeof CAPACIDADES)[number];

export interface CapacidadMeta {
  slug: Capacidad;
  label: string;
  descripcion: string;
  eje: string;
}

export const CAPACIDADES_META: Record<Capacidad, CapacidadMeta> = {
  catalogo: {
    slug: "catalogo",
    label: "Catálogo",
    descripcion: "Productos/servicios con precios, variantes y fotos.",
    eje: "venta",
  },
  agenda: {
    slug: "agenda",
    label: "Agenda",
    descripcion: "Reserva de horas, citas y turnos con recursos/profesionales.",
    eje: "servicio",
  },
  cotizador: {
    slug: "cotizador",
    label: "Cotizador",
    descripcion: "Presupuestos configurables y su aprobación.",
    eje: "erp",
  },
  pedidos: {
    slug: "pedidos",
    label: "Pedidos",
    descripcion: "Órdenes/comandas/órdenes de servicio y su preparación.",
    eje: "operacion",
  },
  pagos: {
    slug: "pagos",
    label: "Pagos",
    descripcion: "Cobro con links, tarjetas y POS. Incluye caja: apertura/arqueo/cierre por turno.",
    eje: "dinero",
  },
  delivery: {
    slug: "delivery",
    label: "Delivery",
    descripcion: "Despacho, reparto y seguimiento de entregas.",
    eje: "logistica",
  },
  suscripciones: {
    slug: "suscripciones",
    label: "Suscripciones",
    descripcion: "Planes recurrentes, membresías, cobros automáticos.",
    eje: "dinero",
  },
  soporte: {
    slug: "soporte",
    label: "Soporte",
    descripcion: "Tickets, casos, mesa de ayuda con SLA. Incluye garantías/RMA post-venta.",
    eje: "crm",
  },
  proyectos: {
    slug: "proyectos",
    label: "Proyectos",
    descripcion: "Trabajos por fases, hitos y avance (incluye obras).",
    eje: "erp",
  },
  inventario: {
    slug: "inventario",
    label: "Inventario",
    descripcion: "Stock, bodegas y movimientos de existencias (producto terminado).",
    eje: "erp",
  },
  compras: {
    slug: "compras",
    label: "Compras",
    descripcion: "Órdenes de compra, proveedores, recepción de mercadería y cuentas por pagar.",
    eje: "erp",
  },
  produccion: {
    slug: "produccion",
    label: "Producción",
    descripcion: "Recetas/BOM, consumo de insumos, órdenes de producción y costeo.",
    eje: "erp",
  },
  expedientes: {
    slug: "expedientes",
    label: "Expedientes",
    descripcion: "Fichas e historial de cliente/paciente/caso.",
    eje: "crm_salud",
  },
  contratos: {
    slug: "contratos",
    label: "Contratos",
    descripcion: "Documentos, firma y control de vigencias.",
    eje: "erp_legal",
  },
  cursos: {
    slug: "cursos",
    label: "Cursos",
    descripcion: "Programas, clases, inscripción y asistencia.",
    eje: "educacion",
  },
  arriendos: {
    slug: "arriendos",
    label: "Arriendos",
    descripcion: "Alquiler de activos y calendario de disponibilidad.",
    eje: "erp",
  },
  facturacion: {
    slug: "facturacion",
    label: "Facturación",
    descripcion:
      "Documentos tributarios (boleta/factura/DTE, notas de crédito). Incluye cuentas por cobrar/crédito B2B y facturación previsional salud (bono/reembolso).",
    eje: "erp",
  },
  fidelizacion: {
    slug: "fidelizacion",
    label: "Fidelización",
    descripcion: "Puntos, sellos, niveles y recompensas.",
    eje: "crm",
  },
  campanas: {
    slug: "campanas",
    label: "Campañas",
    descripcion: "Difusión y mensajería masiva a segmentos.",
    eje: "crm",
  },
  terreno: {
    slug: "terreno",
    label: "Terreno",
    descripcion: "Visitas, rutas y técnicos en terreno.",
    eje: "erp",
  },
  mesas: {
    slug: "mesas",
    label: "Mesas",
    descripcion: "Gestión de mesas, aforo y sala.",
    eje: "gastronomia",
  },
  consentimientos: {
    slug: "consentimientos",
    label: "Consentimientos",
    descripcion: "Formularios y consentimientos firmados.",
    eje: "salud_legal",
  },
};

File: backend/shared/rubros/capacidades.py
"""Catálogo de capacidades (ERP+CRM) — categorización nueva de los rubros.

22 capacidades = las 20 del artifact "Sidebar · 3 propuestas para 100+ rubros" + 2
nuevas ERP (``compras``, ``produccion``) decididas en Fase 0 (vacíos ERP §1.3 del
superprompt). La capacidad es la unidad de la categorización ``rubro → capacidades``
y del gating del árbol de navegación; es agnóstica de la taxonomía de sidebar.

El campo ``capacidades`` STORED en cada ``Rubro`` es el canónico (curado por rubro,
Fase 2). La derivación §3 desde los módulos viejos cumplió su rol de semilla y se
retiró junto con ``modulos`` en la Etapa B (los modos de dato `recurso`/`variantes`/
`precio_medida` viven como flags del ``Rubro``).

Puro (sin ORM), igual que ``diccionario.py``.
"""

from __future__ import annotations

from dataclasses import dataclass
from types import MappingProxyType
from typing import Mapping


class Capacidad:
    """Slugs canónicos de las 22 capacidades (ERP+CRM)."""

    CATALOGO = "catalogo"
    AGENDA = "agenda"
    COTIZADOR = "cotizador"
    PEDIDOS = "pedidos"
    PAGOS = "pagos"
    DELIVERY = "delivery"
    SUSCRIPCIONES = "suscripciones"
    SOPORTE = "soporte"
    PROYECTOS = "proyectos"
    INVENTARIO = "inventario"
    COMPRAS = "compras"
    PRODUCCION = "produccion"
    EXPEDIENTES = "expedientes"
    CONTRATOS = "contratos"
    CURSOS = "cursos"
    ARRIENDOS = "arriendos"
    FACTURACION = "facturacion"
    FIDELIZACION = "fidelizacion"
    CAMPANAS = "campanas"
    TERRENO = "terreno"
    MESAS = "mesas"
    CONSENTIMIENTOS = "consentimientos"
    CONTABILIDAD = "contabilidad"
    ACTIVOS_FIJOS = "activos_fijos"
    RRHH = "rrhh"


# Orden canónico (CAP_ORDER): las tuplas `capacidades` de cada rubro se guardan en
# este orden en AMBOS SSOT (Py y TS) para que el espejo sea determinista.
CAPACIDADES: tuple[str, ...] = (
    Capacidad.CATALOGO,
    Capacidad.AGENDA,
    Capacidad.COTIZADOR,
    Capacidad.PEDIDOS,
    Capacidad.PAGOS,
    Capacidad.DELIVERY,
    Capacidad.SUSCRIPCIONES,
    Capacidad.SOPORTE,
    Capacidad.PROYECTOS,
    Capacidad.INVENTARIO,
    Capacidad.COMPRAS,
    Capacidad.PRODUCCION,
    Capacidad.EXPEDIENTES,
    Capacidad.CONTRATOS,
    Capacidad.CURSOS,
    Capacidad.ARRIENDOS,
    Capacidad.FACTURACION,
    Capacidad.FIDELIZACION,
    Capacidad.CAMPANAS,
    Capacidad.TERRENO,
    Capacidad.MESAS,
    Capacidad.CONSENTIMIENTOS,
    Capacidad.CONTABILIDAD,
    Capacidad.ACTIVOS_FIJOS,
    Capacidad.RRHH,
)


@dataclass(frozen=True)
class CapacidadMeta:
    """Metadata de presentación de una capacidad (label, descripción, eje)."""

    slug: str
    label: str
    descripcion: str
    eje: str


CAPACIDADES_META: Mapping[str, CapacidadMeta] = MappingProxyType(
    {
        Capacidad.CATALOGO: CapacidadMeta(
            Capacidad.CATALOGO,
            "Catálogo",
            "Productos/servicios con precios, variantes y fotos.",
            "venta",
        ),
        Capacidad.AGENDA: CapacidadMeta(
            Capacidad.AGENDA,
            "Agenda",
            "Reserva de horas, citas y turnos con recursos/profesionales.",
            "servicio",
        ),
        Capacidad.COTIZADOR: CapacidadMeta(
            Capacidad.COTIZADOR,
            "Cotizador",
            "Presupuestos configurables y su aprobación.",
            "erp",
        ),
        Capacidad.PEDIDOS: CapacidadMeta(
            Capacidad.PEDIDOS,
            "Pedidos",
            "Órdenes/comandas/órdenes de servicio y su preparación.",
            "operacion",
        ),
        Capacidad.PAGOS: CapacidadMeta(
            Capacidad.PAGOS,
            "Pagos",
            "Cobro con links, tarjetas y POS. Incluye caja: apertura/arqueo/cierre por turno.",
            "dinero",
        ),
        Capacidad.DELIVERY: CapacidadMeta(
            Capacidad.DELIVERY,
            "Delivery",
            "Despacho, reparto y seguimiento de entregas.",
            "logistica",
        ),
        Capacidad.SUSCRIPCIONES: CapacidadMeta(
            Capacidad.SUSCRIPCIONES,
            "Suscripciones",
            "Planes recurrentes, membresías, cobros automáticos.",
            "dinero",
        ),
        Capacidad.SOPORTE: CapacidadMeta(
            Capacidad.SOPORTE,
            "Soporte",
            "Tickets, casos, mesa de ayuda con SLA. Incluye garantías/RMA post-venta.",
            "crm",
        ),
        Capacidad.PROYECTOS: CapacidadMeta(
            Capacidad.PROYECTOS,
            "Proyectos",
            "Trabajos por fases, hitos y avance (incluye obras).",
            "erp",
        ),
        Capacidad.INVENTARIO: CapacidadMeta(
            Capacidad.INVENTARIO,
            "Inventario",
            "Stock, bodegas y movimientos de existencias (producto terminado).",
            "erp",
        ),
        Capacidad.COMPRAS: CapacidadMeta(
            Capacidad.COMPRAS,
            "Compras",
            "Órdenes de compra, proveedores, recepción de mercadería y cuentas por pagar.",
            "erp",
        ),
        Capacidad.PRODUCCION: CapacidadMeta(
            Capacidad.PRODUCCION,
            "Producción",
            "Recetas/BOM, consumo de insumos, órdenes de producción y costeo.",
            "erp",
        ),
        Capacidad.EXPEDIENTES: CapacidadMeta(
            Capacidad.EXPEDIENTES,
            "Expedientes",
            "Fichas e historial de cliente/paciente/caso.",
            "crm_salud",
        ),
        Capacidad.CONTRATOS: CapacidadMeta(
            Capacidad.CONTRATOS,
            "Contratos",
            "Documentos, firma y control de vigencias.",
            "erp_legal",
        ),
        Capacidad.CURSOS: CapacidadMeta(
            Capacidad.CURSOS,
            "Cursos",
            "Programas, clases, inscripción y asistencia.",
            "educacion",
        ),
        Capacidad.ARRIENDOS: CapacidadMeta(
            Capacidad.ARRIENDOS,
            "Arriendos",
            "Alquiler de activos y calendario de disponibilidad.",
            "erp",
        ),
        Capacidad.FACTURACION: CapacidadMeta(
            Capacidad.FACTURACION,
            "Facturación",
            "Documentos tributarios (boleta/factura/DTE, notas de crédito). Incluye "
            "cuentas por cobrar/crédito B2B y facturación previsional salud (bono/reembolso).",
            "erp",
        ),
        Capacidad.FIDELIZACION: CapacidadMeta(
            Capacidad.FIDELIZACION,
            "Fidelización",
            "Puntos, sellos, niveles y recompensas.",
            "crm",
        ),
        Capacidad.CAMPANAS: CapacidadMeta(
            Capacidad.CAMPANAS,
            "Campañas",
            "Difusión y mensajería masiva a segmentos.",
            "crm",
        ),
        Capacidad.TERRENO: CapacidadMeta(
            Capacidad.TERRENO,
            "Terreno",
            "Visitas, rutas y técnicos en terreno.",
            "erp",
        ),
        Capacidad.MESAS: CapacidadMeta(
            Capacidad.MESAS,
            "Mesas",
            "Gestión de mesas, aforo y sala.",
            "gastronomia",
        ),
        Capacidad.CONSENTIMIENTOS: CapacidadMeta(
            Capacidad.CONSENTIMIENTOS,
            "Consentimientos",
            "Formularios y consentimientos firmados.",
            "salud_legal",
        ),
        Capacidad.CONTABILIDAD: CapacidadMeta(
            Capacidad.CONTABILIDAD,
            "Contabilidad",
            "Plan de cuentas, asientos, libro mayor, CxC/CxP, conciliación y estados financieros.",
            "erp",
        ),
        Capacidad.ACTIVOS_FIJOS: CapacidadMeta(
            Capacidad.ACTIVOS_FIJOS,
            "Activos fijos",
            "Registro, depreciación, mantenciones y bajas de activos fijos.",
            "erp",
        ),
        Capacidad.RRHH: CapacidadMeta(
            Capacidad.RRHH,
            "Personas (RRHH)",
            "Empleados, contratos, asistencia, vacaciones, liquidaciones y remuneraciones.",
            "erp",
        ),
    }
)
